Signal/src/ViewControllers/OWSConversationSettingsTableViewController.m

```diff
@@ -115,6 +120,58 @@ - (BOOL)isGroupThread
 - (void)configureWithThread:(TSThread *)thread
 {
     self.thread = thread;
+    [self updateEditButton];
+}
+
+- (void)updateEditButton
+{
+    OWSAssert(self.thread);
+
+    if ([self.thread isKindOfClass:[TSContactThread class]]) {
+        self.navigationItem.rightBarButtonItem =
+            [[UIBarButtonItem alloc] initWithTitle:NSLocalizedString(@"EDIT_TXT", nil)
+                                             style:UIBarButtonItemStylePlain
+                                            target:self
+                                            action:@selector(didTapEditButton)];
+    }
+}
+
+- (void)didTapEditButton
+{
+    if (![self.thread isKindOfClass:[TSContactThread class]]) {
+        DDLogError(@"%@ unexpected thread: %@ in %s", self.tag, self.thread, __PRETTY_FUNCTION__);
+        OWSAssert(NO);
+        return;
+    }
+
+    TSContactThread *contactThread = (TSContactThread *)self.thread;
+    [self.contactsViewHelper presentContactViewControllerForRecipientId:contactThread.contactIdentifier
+                                                     fromViewController:self
+                                                        editImmediately:YES];
```

Comment:
In "group" mode, this view shows the edit view if the thread name or avatar is tapped. Shall we do the same here?